go/vt/schemadiff/capability.go

```diff
@@ -110,13 +113,31 @@ func alterOptionCapableOfInstantDDL(alterOption sqlparser.AlterOption, createTab
 		if tableIsCompressed {
 			return false, nil
 		}
+		if len(createTable.TableSpec.Columns)+len(opt.Columns) > maxColumnsForInstantAddColumn {
```

Comment:
nitpicking: no tests found

Reply:
Good nit! Added unit test and fixed the logic (because obviously there was a bug).